virtual_ecosystem/models/hydrology/hydrology_model.py

```diff
@@ -295,16 +307,50 @@ def _setup(
                 coords={"cell_id": self.grid.cell_id},
             )
 
-        # Set initial aerodynamic resistance for surface, [s m-1]
-        self.data["aerodynamic_resistance_surface"] = DataArray(
-            np.full_like(
-                self.data["elevation"],
-                self.model_constants.initial_aerodynamic_resistance_surface,
-            ),
-            dims="cell_id",
-            name=var,
-            coords={"cell_id": self.grid.cell_id},
+        # Set initial aerodynamic resistance for surface and canopy , [s m-1]
+        self.data["aerodynamic_resistance_surface"] = (
+            self.layer_structure.from_template()
         )
+        self.data["aerodynamic_resistance_surface"][
+            self.layer_structure.index_surface_scalar
+        ] = self.model_constants.initial_aerodynamic_resistance_surface
+
+        self.data["aerodynamic_resistance_canopy"] = (
+            self.layer_structure.from_template()
+        )
+        self.data["aerodynamic_resistance_canopy"][
+            self.layer_structure.index_filled_canopy
+        ] = self.model_constants.initial_aerodynamic_resistance_canopy
+
+        specific_heat_air = abiotic_tools.calculate_specific_heat_air(
+            temperature=self.data["air_temperature"][
+                self.layer_structure.index_filled_atmosphere
+            ].to_numpy(),
+            molar_heat_capacity_air=self.core_constants.molar_heat_capacity_air,
+            specific_heat_equ_factors=self.abiotic_constants.specific_heat_equ_factors,
+        )
+        self.data["specific_heat_air"] = self.layer_structure.from_template()
+        self.data["specific_heat_air"][self.layer_structure.index_filled_atmosphere] = (
+            specific_heat_air
+        )
+
+        # TODO these two variables should probably be initialised elsewhere
+        self.data["stomatal_conductance"] = self.layer_structure.from_template()
+        self.data["stomatal_conductance"][self.layer_structure.index_filled_canopy] = (
+            1000.0
+        )
```

Comment:
@davidorme is that already an output from the plants or could I add it to be initialized by plants?